src/Native/Unix/System.Security.Cryptography.Native/pal_asn1_print.cpp

```diff
@@ -31,12 +31,22 @@ static_assert(PAL_ASN1_STRFLGS_UTF8_CONVERT == ASN1_STRFLGS_UTF8_CONVERT, "");
 
 extern "C" ASN1_STRING* CryptoNative_DecodeAsn1TypeBytes(const uint8_t* buf, int32_t len, Asn1StringTypeFlags type)
 {
+    // This code is never called. Lets just make it compile with all OpenSSL
+    // versions, but only work with OpenSSL 1.0.x.
+
+#if OPENSSL_VERSION_NUMBER < OPENSSL_VERSION_1_1_0_RTM
```

Comment:
This change is not in `master`; the entire file was deleted. Does this make sense?

Reply:
I think this would be

```C++
#if NEED_OPENSSL_1_0_0

    if (!API_EXISTS(d2i_ASN1_type_bytes) || !buf || !len)
    {
        return nullptr;
    }

    return d2i_ASN1_type_bytes(nullptr, &buf, len, type);

#else

    (void)buf;
    (void)len;
    (void)type;
    return nullptr;

#endif
```

Since right now it would SIGSEGV on a portable build that bound against 1.1 and got called by a legacy managed caller (because it was declared LEGACY_FUNCTION, so is null on the 1.1-bound portable).

Since the legacy managed caller handles the `nullptr` return by clearing the error queue and returning C#'s `null` there's no benefit in doing anything more complicated here on the other cases.

Reply:
Thanks!

Reply:
BTW If there's anything else you had to add as LEGACY_FUNCTION it would similarly need an API_EXISTS check.